config/v3_1_experimental/types/headers.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/coreos/ignition/v2/config/shared/errors"
+)
+
+// Parse generates standard net/http headers from the data in HTTPHeaders
+func (h HTTPHeaders) Parse() (http.Header, error) {
+	headers := http.Header{}
+	for _, headerData := range h {
+		if len(headerData) != 2 {
+			return nil, errors.ErrInvalidHTTPHeader
```

Comment:
Optional: hmm, I think we could instead panic here since it shouldn't ever happen if we've passed schema validation already.

Reply:
I do prefer keeping the explicit check plus the error-ing logic. This is a type exposed in the public API which can be directly constructed by consumers.